### DISCUSSION TITLE
Bulk `Link` creation [Jun 07]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2373] [Discord Threads]
> 🙋‍♂️ **WM31**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### <@535455522435104798>, API to manage links currently is one to one. We will consider this as an enhancement request.  
#  
> 🙋‍♂️ **WM31**  
#### You mean there is a way to edit a one row to one row link between table through API?  
#  
> 🙋‍♂️ **WM31**  
#### I am trying to build a master key table to enable that I can do nesting lookup between tables  
#  
> 👨‍🦲 **dstala**  
#### here in this example, I am creating linear links (one-to-one) between two tables using API. Note that I have used 'has-many' as link relation type.  
[1.js](https://cdn.discordapp.com/attachments/983217550177493042/983256080614297610/1.js)  
#  
> 🙋‍♂️ **WM31**  
#### I noticed that the example is done on table creation, is there a way to do so after imports? Like do so through a REST client?  
#  
> 🙋‍♂️ **WM31**  
#### MY current setting is one to many, though I use them for one to one  
#  
> 👨‍🦲 **dstala**  
#### https://apis.nocodb.com/#tag/DB-table-row/operation/db-table-row-nested-add  
#  
> 👨‍🦲 **dstala**  
#### You will need {rowId} & {referenceRowId} :: primary key of two tables that you wish to link.  
#  
> 🙋‍♂️ **WM31**  
#### This page can't help as it don't have the Json sample. can you use your JS sample to demo the Json content to me?  
#  
> 👨‍💻 **pranavxc**  
#### Above api doesn't need any JSON payload instead you have to pass both row id in url path param  
#  
> 👨‍💻 **pranavxc**  
#### ```http://localhost:8080/api/v1/db/data/{orgs}/{projectName}/{tableName}/{rowId}/{relationType}/{columnName}/{refRowId}```  
#  
> 🙋‍♂️ **WM31**  
#### O ok  
#  
> 🙋‍♂️ **WM31**  
#### What is the key for{relation type}?  
#  
> 🙋‍♂️ **WM31**  
#### And both row ID means the system internal row ID? Can those be exported as CSV?  
#  
> 👨‍💻 **pranavxc**  
#### based on the relation created ... it can be `mm` (Many to many) , `hm` (has many) or `bt` (belongs to)  
#  
> 🙋‍♂️ **WM31**  
#### Ok noted on the key  
#  
> 👨‍💻 **pranavxc**  
#### even if you passed wrong relation type it will work  
#  
> 🙋‍♂️ **WM31**  
#### This is a patch or post?  
#  
> 👨‍💻 **pranavxc**  
#### You can enable system fields under `fields` dropdown and the id you can enable  
#  
> 👨‍💻 **pranavxc**  
####   
![image0](https://cdn.discordapp.com/attachments/983217550177493042/983267209918967808/Screenshot_2022-06-06_at_12.42.18_PM.png)  
![image1](https://cdn.discordapp.com/attachments/983217550177493042/983267210170621982/Screenshot_2022-06-06_at_12.42.43_PM.png)  
#  
> 🙋‍♂️ **WM31**  
#### Ok that's helpful  
#  
> 🙋‍♂️ **WM31**  
#### And from the API doc this request is a post, will try myself  
#  
> 🙋‍♂️ **WM31**  
#### But after adding this relation this will be shown in the table view as a column?  
#  
> 👨‍💻 **pranavxc**  
#### If relation is not created yet , you have to create it first  
#  
> 👨‍💻 **pranavxc**  
####   
![image0](https://cdn.discordapp.com/attachments/983217550177493042/983268676868378634/Screenshot_2022-06-06_at_12.48.33_PM.png)  
#  
> 🙋‍♂️ **WM31**  
#### Ok create the relation first then execute the one to one API call  
#  
> 👨‍💻 **pranavxc**  
#### yes  
#  
> 🙋‍♂️ **WM31**  
#### One of topic question, will there be a one click solution to convert excel date to Date value for DB? Or this will be a new feature?  
#  
> 👨‍💻 **pranavxc**  
#### in the api you have to replace the `{columnName}` with alias you had used while creating LinkToAnotherRecord column  
#  
> 👨‍💻 **pranavxc**  
#### In excel import ?  
#  
> 🙋‍♂️ **WM31**  
#### O  
#  
> 🙋‍♂️ **WM31**  
#### I converted that to CSV, so that it becomes a value  
#  
> 🙋‍♂️ **WM31**  
#### I know I can convert that to text before export, but if there is an easier way, the better  
#  
> 👨‍💻 **pranavxc**  
#### you can try to import excel file itself  
#  
> 🙋‍♂️ **WM31**  
#### This means excel import module can handle this, ok will try next time  
#  
> 🙋‍♂️ **WM31**  
#### I just exported it to can to ensure there are no formula and generated values  
#  
> 👨‍💻 **pranavxc**  
#### It will create a new table based on column types and imports the data  
![image0](https://cdn.discordapp.com/attachments/983217550177493042/983270351486214154/Screenshot_2022-06-06_at_12.54.39_PM.png)  
#  
> 🙋‍♂️ **WM31**  
#### Thanks  
#  
> 🙋‍♂️ **WM31**  
#### For API update on date field, ISO format is accepted?  
#  
> 👨‍💻 **pranavxc**  
#### For date we suggest to use `YYYY-MM-DD` format  
#  
> 👨‍💻 **pranavxc**  
#### In our ui we are using the above format  
#  
> 🙋‍♂️ **WM31**  
#### Ok thanks for your support that's all for now🙏  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/983217550177493042)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)

**Comment by review-india** [comment]
https://chick-fil-a-info.com/
